node/node.go

```diff
@@ -371,6 +344,20 @@ func (n *Node) GetESDTData(address, tokenID string, nonce uint64) (*esdt.ESDigit
 		return nil, err
 	}
 
+	if core.ESDTType(esdtToken.Type) != core.Fungible {
+		if esdtToken.TokenMetaData == nil {
+			// TODO: analyze why n.getSystemAccount() doesn't work here. The difference is that here we use AccountsAdapter
+			// and in that function AccountsAdapterAPI is used
+
+			systemAccount, err := n.getAccountHandlerForPubKey(core.SystemAccountAddress)
```

Comment:
you should use the provided interface which know from where to take the NFT data.

Reply:
It would be better to create a NewEsdtDataStorage from vmcommon. Giving him the accountsAPIAdapter and make all reads from there.

Reply:
added